third_party/terraform/resources/resource_storage_bucket.go

```diff
@@ -595,6 +595,29 @@ func resourceStorageBucketDelete(d *schema.ResourceData, meta interface{}) error
 	// Get the bucket
 	bucket := d.Get("name").(string)
 
+	// Try to delete the bucket without clearing it first
```

Comment:
Sorry for the drive-by! Instead of adding a separate deletion here, would it make sense to suppress `403` errors when trying to list bucket contents, and short circuit to the regular deletion call?

That models the list stuff as enhanced behaviour when the user has list permission (and all the list / force_destroy stuff is basically a `pre_delete` piece of custom code)

Reply:
Good idea, done.